src/darwin/Framework/CHIP/MTRFabricInfo.h

```diff
@@ -0,0 +1,101 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * An object that represents a fabric that the MTRDeviceControllerFactory knows
+ * about (and hence one that could have a controller created for it via
+ * createControllerOnExistingFabric).
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+MTR_NEWLY_AVAILABLE
+@interface MTRFabricInfo : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * Root public key for the fabric.
+ */
+@property (readonly, nonatomic) NSData * rootPublicKey MTR_NEWLY_AVAILABLE;
```

Comment:
There shouldn't be a need to annotate each property with availability separately? Any property without an explicit annotation should default to being available whenever the class itself became available.